Updating to web-greeter ^3.0.0

JezerM opened the pull request:
# web-greeter ^3.0.0

A lots of these kind of themes are outdated, depending of an abandoned project: `lightdm-webkit2-greeter`, later renamed to `web-greeter`, made by Antergos team. However, Antergos left some updates and a little direction to what this project should evolve.

So, I decided to fork `web-greeter` to fix and complete this project, with some others new features.

You can check my fork [here](https://github.com/JezerM/web-greeter) (web-greeter 3.1.1), and the API docs [here](https://jezerm.github.io/web-greeter/).

## Litarvan

Litarvan theme is nice! But it does not work at all in this new greeter, because of the use of some deprecated methods or concepts. Therefore, I made some bugfixes to make it usable with `web-greeter ^3.0.0`

Litarvan commented:
This is insane! It would be nice however if the theme could still be retro-compatible with the old web-greeter, so that the update doesn't break on current setups

JezerM commented:
I tried to run **lightdm-webkit2-greeter** in arch to fix it, and I guess it's done~

Works on **nody-greeter**, **web-greeter**, and **lightdm-webkit2-greeter**

## Changes

- Added check for `background_images_dir` and deprecated `background_images`
- Fixed `recDirList` by checking `theme_utils.dirlist`:
    - If `dirlist` returns an Array, it's **lightdm-webkit2-greeter**
    - Else, it resolves with the callback function.
- Check if `lightdm.language` is string (deprecated), or a LightDMLanguage object.
